src/thumbnailer_slope_optim.cc

```diff
@@ -18,16 +18,17 @@ namespace libwebp {
 
 Thumbnailer::Status Thumbnailer::GenerateAnimationSlopeOptim(
     WebPData* const webp_data) {
-  Thumbnailer::Status anim_status = LossyEncodeSlopeOptim(webp_data);
-  if (anim_status == kOk) {
-    anim_status = TryNearLossless(webp_data);
-  }
+  CHECK_THUMBNAILER_STATUS(LossyEncodeSlopeOptim(webp_data));
+  CHECK_THUMBNAILER_STATUS(TryNearLossless(webp_data));
 
-  if (anim_status == kOk) {
-    anim_status = LossyEncodeNoSlopeOptim(webp_data);
-  }
+  int curr_anim_size = webp_data->size;
+  do {
+    CHECK_THUMBNAILER_STATUS(LossyEncodeNoSlopeOptim(webp_data));
+    if (curr_anim_size == webp_data->size) break;
+    curr_anim_size = webp_data->size;
+  } while (0);
```

Comment:
What is this while for?

Reply:
the `while` loop is for guarantee that we cannot get better results with `LossyEncodeNoSlopeOptim` anymore.

Reply:
This is a while(0) so it will only run once. If it was a while(true), you would need some guarantee that it will stop.

Reply:
It's my mistake, I thought that `while(true)` and `while(0)` were similar :) .